server/src/views/profile.js

```diff
@@ -0,0 +1,249 @@
+/* jshint browser:true */
+
+const React = require("react");
+const errors = require("../../shared/errors");
+
+function delay(duration) {
+  return new Promise(resolve => setTimeout(resolve, duration));
+}
+
+exports.ProfileButton = class ProfileButton extends React.Component {
+  constructor(props) {
+    super(props);
+    this.onDocumentClick = this.dismiss.bind(this);
+    this.onRefreshProfileBound = this.onRefreshProfile.bind(this);
+    this.state = {
```

Comment:
avatar, nickname, and email can probably all be props, while isExpanded and isFetching are probably right to have as state.  I think `getInitialState` is usually how that state is set